x-pack/test/apm_api_integration/tests/asset_services/asset_services.spec.ts

```diff
@@ -0,0 +1,239 @@
+          instance
+            .transaction({ transactionName })
+            .timestamp(timestamp)
+            .duration(1000)
+            .failure()
+            .errors(
+              instance
+                .error({
+                  message: '[ResponseError] index_not_found_exception',
+                  type: 'ResponseError',
+                })
+                .timestamp(timestamp + 50)
+            )
+        );
+
+        const metricsets = range
+          .interval('30s')
+          .rate(1)
+          .generator((timestamp) =>
+            instance
+              .appMetrics({
+                'system.memory.actual.free': 800,
+                'system.memory.total': 1000,
+                'system.cpu.total.norm.pct': 0.6,
+                'system.process.cpu.total.norm.pct': 0.7,
+              })
+              .timestamp(timestamp)
+          );
+
+        return [...successfulTraceEvents, ...failedTraceEvents, ...metricsets];
+      };
+
+      const logEvents = range
+        .interval('1m')
+        .rate(1)
+        .generator((timestamp) => {
+          return [
+            log
+              .create()
+              .message('this is a log message')
+              .service('logs-only-service')
+              .logLevel('error')
+              .defaults({
+                'trace.id': generateShortId(),
+                'agent.name': 'synth-agent',
+              })
+              .timestamp(timestamp),
+            log
+              .create()
+              .message('this is a log message')
+              .service('multisignal-service')
+              .logLevel('error')
+              .defaults({
+                'trace.id': generateShortId(),
+                'agent.name': 'synth-agent',
+              })
+              .timestamp(timestamp),
+          ];
+        });
+
+      function* createGeneratorFromArray(arr: Array<Serializable<any>>) {
+        yield* arr;
+      }
+      const traces = instances.flatMap((instance) => instanceSpans(instance));
+      const tracesGenAssets = createGeneratorFromArray(traces);
+
+      //
+      return await assetsSynthtraceClient.index(
+        Readable.from(Array.from(logEvents).concat(Array.from(tracesGenAssets)))
+      );
+    });
+
+    after(async () => {
+      await assetsSynthtraceClient.clean();
+    });
+
+    describe('when no additional filters are applied', () => {
+      before(async () => {
+        response = await callApi({ start, end, kuery: '' });
+      });
+
+      it('returns a successful response', async () => {
+        expect(response.status).to.be(200);
+        expect(response.body.services.length).to.be(3);
+      });
+
+      it('returns all services', () => {
+        const serviceNames = response.body.services.map((item) => item.service.name);
+
+        expect(serviceNames).to.contain('multisignal-service');
+        expect(serviceNames).to.contain('apm-only-service');
+        expect(serviceNames).to.contain('logs-only-service');
```

Comment:
Should we also check the signType here?

Reply:
sure 

https://github.com/elastic/kibana/pull/181842/commits/4f4432fba919e9b3b18a494a957676c47695fde5#diff-d6a4d445aa0be964f83d8b3b7488ef1f4fc0e445a5cd71cd49bd495fa45b5107R193